pkg/arvo/app/metadata-hook.hoon

```diff
@@ -0,0 +1,200 @@
+::  metadata-hook: allow syncing foreign metadata
+::
+/-  *metadata-store, *metadata-hook
+/+  default-agent
+|%
++$  card  card:agent:gall
++$  versioned-state
+  $%  state-zero
+  ==
+::
++$  state-zero
+  $:  %0
+      synced=(map group-path ship)
+  ==
+--
+=|  state-zero
+=*  state  -
+^-  agent:gall
+=<
+  |_  =bowl:gall
+  +*  this        .
+      hc          ~(. +> bowl)
+      def         ~(. (default-agent this %|) bowl)
+  ::
+  ++  on-init
+    [[%pass /updates %agent [our.bowl %metadata-store] %watch /updates]~ this]
+  ::
+  ++  on-save   !>(state)
+  ++  on-load   |=(=vase `this(state !<(state-zero vase)))
+  ++  on-leave  on-leave:def
+  ++  on-peek   on-peek:def
+  ++  on-arvo   on-arvo:def
+  ++  on-fail   on-fail:def
+  ++  on-poke
+    |=  [=mark =vase]
+    ^-  (quip card _this)
+    ?+  mark  (on-poke:def mark vase)
+        %metadata-hook-action
+      =^  cards  state
+        (poke-hook-action:hc !<(metadata-hook-action vase))
+      [cards this]
+    == 
+  ::
+  ++  on-watch
+    |=  =path
+    ^-  (quip card _this)
+    ?+  path        (on-watch:def path)
+        [%group *]  [(watch-group:hc t.path) this]
+    ==
+  ::
+  ++  on-agent
+    |=  [=wire =sign:agent:gall]
+    ^-  (quip card _this)
+    ?+  -.sign  (on-agent:def wire sign)
+        %kick       =^(cards state (kick:hc wire) [cards this])
+        %watch-ack  =^(cards state (watch-ack:hc wire p.sign) [cards this])
+        %fact
+      ?+  p.cage.sign  (on-agent:def wire sign)
+          %metadata-update
+        =^  cards  state
+          (fact-metadata-update:hc wire !<(metadata-update q.cage.sign))
+        [cards this]
+      ==
+    ==
+  --
+::
+|_  =bowl:gall
+++  poke-hook-action
+  |=  act=metadata-hook-action
+  ^-  (quip card _state)
+  |^
+  ?-  -.act
+      %add-owned
+    ?>  (team:title our.bowl src.bowl)
+    :-  ~
+    ?:  (~(has by synced) path.act)  state
+    state(synced (~(put by synced) path.act our.bowl))
```

Comment:
It's a little bit confusing that `%add-owned` puts items in to `synced` just like `%add-synced` does... but I think that at least follows the pattern used by other existing hooks?

Reply:
I really like this pattern actually, as it keeps track of our "hosted resources" in the same data structure and format as our "synced resources". Ted recommended it when I was first experimenting with hooks, and it really stuck.